ale_linters/rego/opacheck.vim

```diff
@@ -11,7 +11,7 @@ function! ale_linters#rego#opacheck#GetCommand(buffer) abort
     let l:options = ale#Var(a:buffer, 'rego_opacheck_options')
 
     return ale#Escape(ale_linters#rego#opacheck#GetExecutable(a:buffer))
-    \   . ' check %s --format json '
+    \   . ' check . --format json '
```

Comment:
Because the command can start asynchronously, this could fire from a different buffer from the buffer you started on. Luckily, I added a quick way to get the directory of a file with `fnamemodify`-like syntax a while ago.

```suggestion
    \   . ' check %s:h --format json '
```

Try this instead, and let me know if it works.

Reply:
I noted in the documentation for `ale#linter#Define` that by default ALE doesn't set a working directory for commands as that can actually break a lot of things. Another thing you could try is adding `'cwd': '%s:h'` to the arguments below, and then `.` here should work. If you grep the codebase, a lot of linters explicitly set the work directory to be the directory of the file. Setting `cwd` instead may or may not work better here, it's better to just run the linter and see what happens.

Reply:
It's been a while since I've done anything with `opa` or `rego` but I just tried this out. For a file `%s` throws some errors due to missing context from the rest of the directory, while both `.` and `%s:h` seem to work fine.

I can see in `:ALEInfo` that using `%s:h` shows the full path to the directory and that seems simpler to me than setting the cwd first.